src/Util/ActorDimensionKeeper.cpp

```diff
@@ -0,0 +1,71 @@
+#include "ActorDimensionKeeper.h"
+#include <al/Library/LiveActor/ActorPoseKeeper.h>
+#include <container/seadPtrArray.h>
+
+void ActorDimensionKeeper::validate() {
+    mIsValid = true;
+}
+
+void ActorDimensionKeeper::invalidate() {
+    mIsValid = false;
+    if (!mIs2D)
+        return;
+    mIs2D = false;
+    mIsIn2DArea = false;
+    mIsCurrently2D = false;
+    mIsCanChange3D = true;
+}
+
+void ActorDimensionKeeper::forceChange2DKeep() {
+    mIsCurrently2D = true;
+}
+
+void ActorDimensionKeeper::forceEndChange2DKeep() {
+    mIsCurrently2D = false;
+}
+
+void ActorDimensionKeeper::update() {
+    sead::PtrArray<al::LiveActor*> actors;
+    actors.setBuffer(8, nullptr);
+
+    if (mIsValid) {
+        auto actorPos = al::getTrans(mLiveActor);
+    }
```

Comment:
seems weird, but can't verify right now

Reply:
Didn't finish this one it looks like. Removed instead of NON_MATCHING as it's very incomplete.